trifinger_simulation/pinocchio_utils.py

```diff
@@ -27,42 +27,59 @@ def __init__(
             for link_name in tip_link_names
         ]
 
-    def forward_kinematics(self, joint_positions) -> typing.List[np.ndarray]:
-        """Compute end-effector positions for the given joint configuration.
+    def forward_kinematics(
+        self,
+        joint_positions: typing.List[np.ndarray],
+        joint_velocities: typing.Optional[np.ndarray] = None,
+    ) -> typing.Union[
+        typing.List[np.ndarray],
+        typing.Tuple[typing.List[np.ndarray], typing.List[np.ndarray]],
+    ]:
+        """Compute end-effector positions (and velocities) for the given joint configuration.
 
         Args:
             joint_positions:  Flat list of angular joint positions.
+            joint_positions (optional):  Flat list of angular joint
+                velocities.
 
         Returns:
-            List of end-effector positions. Each position is given as an
+            List of end-effector positions and, if join_velocities are provided,
+            end-effector velocities. Each position and velocity is given as an
             np.array with x,y,z positions.
```

Comment:
I think it might be good to clarify that the return type depends on the input.  Something like:

    If only position is given: List of ...
    If velocity is given: Tuple of List of ...

Reply:
Thank you for your feedback. I have changed the docstring as suggested.